python-package/securexgboost/core.py

```diff
@@ -469,397 +529,431 @@ def __init__(self, data, encrypted=False, label=None, missing=None,
+    # 
+    #     Parameters
+    #     ----------
+    #     weight : array like
+    #         Weight for each data point in numpy 2D array
+    # 
+    #         .. note:: For ranking task, weights are per-group.
+    # 
+    #             In ranking task, one weight is assigned to each group (not each data
+    #             point). This is because we only care about the relative ordering of
+    #             data points within each group, so it doesn't make sense to assign
+    #             weights to individual data points.
+    #     """
+    #     self.set_float_info_npy2d('weight', weight)
+
+    # TODO(rishabh): Enable this API with encryption
+    # def set_base_margin(self, margin):
+    #     """ Set base margin of booster to start from.
+    # 
+    #     This can be used to specify a prediction value of
+    #     existing model to be base_margin
+    #     However, remember margin is needed, instead of transformed prediction
+    #     e.g. for logistic regression: need to put in value before logistic transformation
+    #     see also example/demo.py
+    # 
+    #     Parameters
+    #     ----------
+    #     margin: array like
+    #         Prediction margin of each datapoint
+    #     """
+    #     self.set_float_info('base_margin', margin)
+
+    # TODO(rishabh): Enable this API with encryption
+    # def set_group(self, group):
+    #     """Set group size of DMatrix (used for ranking).
+    # 
+    #     Parameters
+    #     ----------
+    #     group : array like
+    #         Group size of each group
+    #     """
+    #     _check_call(_LIB.XGDMatrixSetGroup(self.handle,
+    #                                        c_array(ctypes.c_uint, group),
+    #                                        c_bst_ulong(len(group))))
+
+    # TODO(rishabh): Enable this API with encryption
+    # def get_label(self):
+    #     """Get the label of the DMatrix.
+    # 
+    #     Returns
+    #     -------
+    #     label : array
+    #     """
+    #     return self.get_float_info('label')
+
+    # TODO(rishabh): Enable this API with encryption
+    # def get_weight(self):
+    #     """Get the weight of the DMatrix.
+    # 
+    #     Returns
+    #     -------
+    #     weight : array
+    #     """
+    #     return self.get_float_info('weight')
+
+    # TODO(rishabh): Enable this API with encryption
+    # def get_base_margin(self):
+    #     """Get the base margin of the DMatrix.
+    # 
+    #     Returns
+    #     -------
+    #     base_margin : float
+    #     """
+    #     return self.get_float_info('base_margin')
+
+    # FIXME: Enable this API with encryption
     def num_row(self):
         """Get the number of rows in the DMatrix.
 
         Returns
         -------
         number of rows : int
         """
-        ret = c_bst_ulong()
-        _check_call(_LIB.XGDMatrixNumRow(self.handle,
-                                         ctypes.byref(ret)))
-        return ret.value
+        if channel_addr:
```

Comment:
Do we need to expose `num_row()` and `num_col`? I saw that they're called in `predict()`, but we commented out that code.

Reply:
I think it's required by feature_names as well, which in turn is required by _validate_features, which is called a bunch of places.

Reply:
If I remember correctly, we can comment out _validate_features everywhere and it won't break anything. But there should be no harm in exposing these two functions, so we can just leave it.